services/resources-getter.js

```diff
@@ -268,7 +266,24 @@ function ResourcesGetter(model, opts, params) {
           }
         }
 
-        return [records, count, fieldsSearched];
+        return [records, fieldsSearched];
+      });
+  };
+
+  this.count = function () {
+    return getSegmentCondition()
+      .then(function () {
+        var query = getRecordsQuery();
+
+        if (hasRelationshipFilter()) {
+          return exec(query)
+            .then(function (records) {
+              var count = records.length;
+              return count;
```

Comment:
`return records.length;` is lighter.

Can you improve this code to prevent the whole records count in Javascript?